### PR TITLE
chore: Adding evaluation version in application for new evaluation logic.

### LINKED ISSUE #2862: [Bug] Modify dynamic binding evaluation to support escaped characters without double escapes
Escape characters (like `\t`, `\n` etc) are currently not read as such during evaluation. They need to be written as a double escaped `\\n` for the character to make sense while evaluation. This is non-intuitive since the editor space is not represented as a string literal.

### PR DESCRIPTION
**Pull request by trishaanand**
Fixes #2862

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ApplicationPageServiceImpl.java [hunk 1]
```diff
@@ -247,6 +249,9 @@ private Boolean isDuplicatePage(Application application, String pageId) {
 
         application.setPublishedPages(new ArrayList<>());
 
+        // For all new applications being created, set it to use the latest evaluation version.
+        application.setEvaluationVersion(EVALUATION_VERSION);
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ApplicationPageServiceImpl.java:253]: @trishaanand uneducated question, is this flow also used for clone applications, fork applications, import/export etc?

Comment (trishaanand) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ApplicationPageServiceImpl.java:253]: Aah yes. Great catch! Let me look at the code and figure out how to differentiate between these flows.

Comment (trishaanand) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ApplicationPageServiceImpl.java:253]: Checked. We are good. All of these flows manually handle the objects for the application and do not use the create application flow <phew> @nidhi-nair

**Comment by trishaanand** [comment]
Test run at : https://github.com/appsmithorg/appsmith/runs/4089854695?check_suite_focus=true

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ApplicationPageServiceImpl.java (hunk 1 above)
Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ApplicationPageServiceImpl.java:253]: Got it, so then version null will be duplicated as version null and version 2 will be duplicated as version 2 automatically?

Comment (trishaanand) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ApplicationPageServiceImpl.java:253]: Thats correct.

**Review by nidhi-nair — APPROVED** [review]

**Pull request merged**